node/src/accountant/scanners.rs

```diff
@@ -329,34 +314,134 @@ impl PayableScanner {
         }
     }
 
-    fn handle_errors(&self, errors: Vec<BlockchainError>, logger: &Logger) {
-        for blockchain_error in errors {
-            if let Some(hash) = blockchain_error.carries_transaction_hash() {
-                if let Some(rowid) = self.pending_payable_dao.fingerprint_rowid(hash) {
-                    debug!(
-                        logger,
-                        "Deleting an existing fingerprint for a failed transaction {:?}", hash
-                    );
-                    if let Err(e) = self.pending_payable_dao.delete_fingerprint(rowid) {
-                        panic!(
-                            "Database unmaintainable; payable fingerprint deletion for \
-                                transaction {:?} has stayed undone due to {:?}",
-                            hash, e
-                        );
-                    };
-                };
-
-                warning!(
+    fn separate_id_triples_from_existent_and_nonexistent_fingerprints<'a>(
+        &'a self,
+        sent_payments: &'a [&'a PendingPayable],
+    ) -> (
+        Vec<RefWalletAndRowidOptCoupledWithHash>,
+        Vec<RefWalletAndRowidOptCoupledWithHash>,
+    ) {
+        let hashes = sent_payments
+            .iter()
+            .map(|pending_payable| pending_payable.hash)
+            .collect::<Vec<H256>>();
+        self.pending_payable_dao
+            .fingerprints_rowids(&hashes)
```

Comment:
Maybe this name should be changed to reflect the order of items in the return value.

Or the order of the items in the return value could be changed.